src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddFileMetadataToDeposit.scala

```diff
@@ -18,77 +18,103 @@ package nl.knaw.dans.easy.multideposit.actions
 import java.io.File
 
 import nl.knaw.dans.easy.multideposit.actions.AddFileMetadataToDeposit._
-import nl.knaw.dans.easy.multideposit.{ UnitAction, Settings, _ }
+import nl.knaw.dans.easy.multideposit.{ Settings, UnitAction, _ }
 import nl.knaw.dans.lib.error.TraversableTryExtensions
 import org.apache.tika.Tika
 
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Success, Try }
+import scala.util.{ Failure, Try }
 import scala.xml.Elem
 
 case class AddFileMetadataToDeposit(row: Int, entry: (DatasetID, Dataset))(implicit settings: Settings) extends UnitAction[Unit] {
 
   val (datasetID, dataset) = entry
 
+  // TODO @rvanheest move the mimetype computation to a separate action once the new
+  // parameterized version of Action is installed?
+  lazy val mimetypeMap: Try[List[(File, String)]] = Try {
+    val datasetDir = multiDepositDir(datasetID)
+    if (datasetDir.exists()) {
+      val files = datasetDir.listRecursively
+
+      files.map(getMimeType)
+        .collectResults
+        .map(files.zip(_))
+    }
+    else // this means the dataset does not contain any data
+      Try { List.empty }
+  }.flatten
+
   /**
    * Verifies whether all preconditions are met for this specific action.
    * All files referenced in the instructions are checked for existence.
    *
    * @return `Success` when all preconditions are met, `Failure` otherwise
    */
   override def checkPreconditions: Try[Unit] = {
-    val inputDir = settings.multidepositDir
-    // Note that the FILE_SIP paths are not really used in this action
-    val nonExistingFileSIPs = dataset.getOrElse("FILE_SIP", List.empty)
-      .filterNot(_.isEmpty)
-      .filterNot(fp => new File(settings.multidepositDir, fp).exists())
-
-    if (nonExistingFileSIPs.isEmpty) Success(())
-    else Failure(ActionException(row, s"""The following SIP files are referenced in the instructions but not found in the deposit input dir "$inputDir" for dataset "$datasetID": ${ nonExistingFileSIPs.mkString("[", ", ", "]") }""".stripMargin))
-  }
+    import validators._
 
-  override def execute(): Try[Unit] = writeFileMetadataXml(row, datasetID)
-}
-object AddFileMetadataToDeposit {
+    def checkSFColumnsIfDatasetContainsAVFiles(mimetypes: List[(File, String)]): Try[Unit] = {
+      val avFiles = mimetypes.filter {
+        case (_, mimetype) => (mimetype startsWith "video") || (mimetype startsWith "audio")
+      }
+
+      if (avFiles.nonEmpty)
+        checkColumnsAreNonEmpty(row, dataset, "SF_USER", "SF_COLLECTION")
+          .recoverWith {
+            case ActionException(r, msg, _) => Failure(ActionException(r,
+              s"$msg\ncause: these columns should contain values because audio/video files are " +
+                s"found:\n${ avFiles.map { case (file, _) => s" - $file" }.mkString("\n") }"))
```

Comment:
Aren't the file right also important whether to create streaming or not? Can't I leave these columns empty if I wan't to archive them but have them accessible to none?